pkg/operators/limiter/limiter.go

```diff
@@ -0,0 +1,164 @@
+// Copyright 2024 The Inspektor Gadget authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package limiter is a data operator that operates on array data sources by
+// limiting the number of entries in each batch of data. A great scenario for
+// this operator is when you are already sorting data within an array of data
+// and you want to filter out the top `X` entries.
+package limiter
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/datasource"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/gadget-service/api"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/operators"
+	"github.com/inspektor-gadget/inspektor-gadget/pkg/params"
+)
+
+const (
+	name            = "limiter"
+	ParamMaxEntries = "max-entries"
+	Priority        = 9600
+)
+
+type limiterOperator struct{}
+
+func (l *limiterOperator) Name() string {
+	return name
+}
+
+func (l *limiterOperator) Init(params *params.Params) error {
+	return nil
+}
+
+func (l *limiterOperator) GlobalParams() api.Params {
+	return nil
+}
+
+func (l *limiterOperator) InstanceParams() api.Params {
+	return api.Params{
+		{
+			Key:   ParamMaxEntries,
+			Title: "Max Entries",
+			Description: "The maximal number of entries for each batch of data. " +
+				"If using multiple data sources, prefix the value with 'datasourcename:' and separate with ';'.",
+			DefaultValue: "50",
```

Comment:
I'm not sure about having this default value - it might be rather surprising for gadget authors if they're not aware of this operator/functionality to have their data limited. I'd suggest keeping it empty (unlimited) or use `-1` explicitly for "unlimited".

Reply:
I also thought about not limiting but I wanted to initially keep the same behaviour we had in built-in gadgets. Let's see what other think also here.

Reply:
I set no limit by default.